packages/product/src/repositories/product-option-value.ts

```diff
@@ -0,0 +1,109 @@
+import { ProductOptionValue } from "@models"
+import { Context, DAL } from "@medusajs/types"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+import { DALUtils } from "@medusajs/utils"
+import {
+  CreateProductOptionValueDTO,
+  UpdateProductOptionValueDTO,
+} from "../types/services/product-option-value"
+import { FilterQuery as MikroFilterQuery } from "@mikro-orm/core/typings"
+import { FindOptions as MikroOptions } from "@mikro-orm/core/drivers/IDatabaseDriver"
+
+export class ProductOptionValueRepository extends DALUtils.MikroOrmBaseRepository {
+  protected readonly manager_: SqlEntityManager
+
+  constructor({ manager }: { manager: SqlEntityManager }) {
+    // @ts-ignore
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments)
+    this.manager_ = manager
+  }
+
+  async find(
+    findOptions: DAL.FindOptions<ProductOptionValue> = { where: {} },
+    context: Context = {}
+  ): Promise<ProductOptionValue[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+    const findOptions_ = { ...findOptions }
+
+    findOptions_.options ??= {}
+
+    return await manager.find(
+      ProductOptionValue,
+      findOptions_.where as MikroFilterQuery<ProductOptionValue>,
+      findOptions_.options as MikroOptions<ProductOptionValue>
+    )
+  }
+
+  async upsert(
+    optionValues: (UpdateProductOptionValueDTO | CreateProductOptionValueDTO)[],
+    context: Context = {}
+  ): Promise<ProductOptionValue[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const optionValueIds = optionValues
+      .filter((option) => !!option.id)
+      .map(({ id }) => id!)
```

Comment:
todo: just thinking that maybe instead of a double loop we could do something like
```ts
const optionValueIds = []
optionValues.forEach(optionValue => {
  if (!!optionValue.id) optionValueIds.push(optionValue.id)
})
```